pkg/opensearch/client/ppl_request.go

```diff
@@ -31,11 +31,17 @@ func (b *PPLRequestBuilder) AddPPLQueryString(timeField, to, from, querystring s
 	var res []string
 	timeFilter := fmt.Sprintf(" where `%s` >= timestamp('%s') and `%s` <= timestamp('%s')", timeField, from, timeField, to)
 
+	trimmedQuerystring := strings.TrimSpace(querystring)
 	// Sets a default query if the query string is empty
-	if len(strings.TrimSpace(querystring)) == 0 {
+	if len(trimmedQuerystring) == 0 {
 		querystring = fmt.Sprintf("source = %s", b.index)
 	}
 
+	// Set a source index if the query string is not empty, but includes ad-hoc filters.
+	if !strings.HasPrefix(trimmedQuerystring, "source") && !strings.HasPrefix(trimmedQuerystring, "search") {
```

Comment:
no real reason. both conditions appeared to be the same to me. i've updated it though to check for "| where `", since that matches what the the frontend does when adding ad-hoc filters